modules/integration/integration-test-utils/src/main/java/org/wso2/appfactory/integration/test/utils/AFDefaultDataPopulator.java

```diff
@@ -507,7 +507,7 @@ public void waitUntilApplicationDeletionCompletes(long waitInterval, int retryCo
 		while (round <= retryCount) {
 			try {
 				httpResponse = appMgtRestClient.getAppInfo(applicationKey, false);
-				if(httpResponse != null && httpResponse.getData().equals("null")) {
+				if(httpResponse != null && "null".equals(httpResponse.getData().trim())) {
```

Comment:
I guess correct way to handle this is to send some error HTTP status from the jaggery side when an error occurred during the application deletion. Then we can check the response status, instead of the response data